lib/std/io.zig

```diff
@@ -603,7 +616,10 @@ pub fn Poller(comptime StreamEnum: type) type {
 
             const err_mask = os.POLL.ERR | os.POLL.NVAL | os.POLL.HUP;
 
-            const events_len = try os.poll(&self.poll_fds, std.math.maxInt(i32));
+            const events_len = try os.poll(&self.poll_fds, if (nanoseconds) |ns|
+                std.math.cast(i32, ns / std.time.ns_per_ms) orelse std.math.maxInt(i32)
+            else
+                -1);
```

Comment:
I changed this to `-1` as https://man7.org/linux/man-pages/man2/poll.2.html indicates:
> Specifying a negative value in timeout means an infinite timeout.

The old behavior was just waiting `std.math.maxInt(i32)` milliseconds it seems.